percy/providers/generic_provider.py

```diff
@@ -60,9 +64,91 @@ def _get_tiles(self, **kwargs):
+
+        return ignore_region
+
+    def ignore_regions_by_xpaths(self, ignored_elements_array, xpaths):
+        for xpath in xpaths:
+            try:
+                element = self.driver.find_element(by=AppiumBy.XPATH, value=xpath)
+                selector = f"xpath: {xpath}"
+                ignored_region = self.ignore_element_object(selector, element)
+                ignored_elements_array.append(ignored_region)
+            except NoSuchElementException as e:
+                log(f"Appium Element with xpath: {xpath} not found. Ignoring this xpath.")
+                log(e, on_debug=True)
+
+    def ignore_regions_by_ids(self, ignored_elements_array, ids):
+        for id in ids:
+            try:
+                element = self.driver.find_element(by=AppiumBy.ACCESSIBILITY_ID, value=id)
+                selector = f"id: {id}"
+                ignored_region = self.ignore_element_object(selector, element)
+                ignored_elements_array.append(ignored_region)
+            except NoSuchElementException as e:
+                log(f"Appium Element with id: {id} not found. Ignoring this id.")
+                log(e, on_debug=True)
+
+    def ignore_regions_by_elements(self, ignored_elements_array, elements):
+        for idx, element in enumerate(elements):
+            try:
+                class_name = element.get_attribute('class')
+                selector = f"element: {idx} {class_name}"
+                ignored_region = self.ignore_element_object(selector, element)
+                ignored_elements_array.append(ignored_region)
+            except NoSuchElementException as e:
+                log(f"Correct Element not passed at index {idx}")
+                log(e, on_debug=True)
+
+    def add_custom_ignore_regions(self, ignored_elements_array, custom_locations):
+        for idx, custom_location in enumerate(custom_locations):
+            screen_width = self.metadata.device_screen_size['width']
+            screen_height = self.metadata.device_screen_size['height']
+            if custom_location.is_valid(screen_height, screen_width):
+                ignored_region = {
+                    "selector": f"custom ignore region: {idx}",
+                    "coOrdinates": {
+                        "top": custom_location.top,
+                        "bottom": custom_location.bottom,
+                        "left": custom_location.left,
+                        "right": custom_location.right
+                    }
+                }
+                ignored_elements_array.append(ignored_region)
+            else:
+                log(f"Values passed in custom ignored region at index: {idx} is not valid")
+
     @staticmethod
-    def _post_screenshots(name, tag, tiles, debug_url):
-        response = CLIWrapper().post_screenshots(name, tag, tiles, debug_url)
+    def _post_screenshots(name, tag, tiles, ignored_regions, debug_url):
+        response = CLIWrapper().post_screenshots(name, tag, tiles, ignored_regions, debug_url)
```

Comment:
here you are passing it as second last arg

Reply:
covered in integration testing